packages/ui/src/domainModules/agent/components/AgentIdTextResult/index.js

```diff
@@ -95,18 +95,23 @@ class AgentIdTextResult extends Component {
       'attributes.disambiguatingDescription'
     )
 
+    let suffix
+    if (!textOnly) {
+      suffix =
+        (fullName && moduleTranslate({ module: 'agent', textKey: 'agent' })) ||
+        (inputText && moduleTranslate({ module: 'form', textKey: 'plainText' }))
+    }
+
     const agentName = fullName || inputText
-    const suffix =
-      (fullName && moduleTranslate({ module: 'agent', textKey: 'agent' })) ||
-      (inputText &&
-        moduleTranslate({
-          module: 'form',
-          textKey: 'plainText',
-        }))
-
-    const agentNameSuffix = disambiguatingDescription
-      ? `(${disambiguatingDescription}) [${suffix}]`
-      : `[${suffix}]`
+
+    let agentNameSuffix
+    if (textOnly) {
+      agentNameSuffix = disambiguatingDescription
+    } else {
+      agentNameSuffix = disambiguatingDescription
+        ? `(${disambiguatingDescription}) [${suffix}]`
+        : `[${suffix}]`
+    }
```

Comment:
this part (lines 98-114) could be simplified like this (mainly because the first `if (!textOnly)` is the same case as `if (textOnly)...else`):

```javascript
const agentName = fullName || inputText

let agentNameSuffix
if (textOnly) {
  agentNameSuffix = disambiguatingDescription
} else {
  const typeSuffix =
    (fullName && moduleTranslate({ module: 'agent', textKey: 'agent' })) ||
    (inputText && moduleTranslate({ module: 'form', textKey: 'plainText' }))

  agentNameSuffix = disambiguatingDescription
    ? `(${disambiguatingDescription}) [${typeSuffix}]`
    : `[${typeSuffix}]`
}
```

Reply:
Right. I will fix it.